src/api/error.rs

```diff
@@ -89,15 +89,15 @@ pub enum ApiError {
     #[error("JSON encoding failed")]
     JsonEncodingFailed(String, serde_json::Error),
 
+    // SDK error.
+    #[error("an unexpected SDK error has occured.")]
+    UnexpectedSdkError,
```

Comment:
I suggest adding a string parameter for an explanatory message (like the underlying error message).

Actually the error should probably be more generic like just "unexpected error" and then the message could add that the error is with the SDK. That way we don't end up with a bunch of error cases for things that never happens.

Reply:
I did something very like this using an `InternalServerError` using `anyhow`. Check the newest changes for details